epistasis_selection.py

```diff
@@ -0,0 +1,98 @@
+import itertools
+from typing import Optional, Iterable, Tuple, List
+
+import numpy as np
+
+import potts_model
+import utils
+
+
+def combine_k_rounds(num_rounds: int, mutations: Iterable[Iterable[Tuple[int, int]]]) -> List[List[Tuple[int, int]]]:
+  """Return the result of combining `mutations` for `num_rounds`.
+
+  Starting with a pool of M `mutations` m_1 ... m_M, stack them for K=`num_rounds` rounds. For example,
+  for K=3 rounds of combination, this will result in every variant (m_i + m_j + m_k), for i, j, k \in M.
+  Be careful of memory usage, as this can be very large due to combinatorial possibilities.
+  In the best case, this scales with {M \choose K}. But if mutations overlap at P positions,
+  combining them produces 2^{P} variants. So in the worst case, this will produce
```

Comment:
I found this confusing. If the mutations overlap, aren't there fewer possibilities than if they don't overlap?

Reply:
got it. makes sense. it wasn't intutivie that the overlaps would lead to *more* diversity, but it makes sense now.